Building tests fails on non-x86 architectures

```rust
error[E0433]: failed to resolve: use of undeclared type or module `KernelAvx`
   --> src/sgemm_kernel.rs:507:27
    |
507 |             Alloc::new(n, KernelAvx::align_to()).init_with(elt)
    |                           ^^^^^^^^^ use of undeclared type or module `KernelAvx`
error: cannot find macro `loop_m!` in this scope
   --> src/sgemm_kernel.rs:542:9
    |
542 |         loop_m!(i, loop_n!(j, m[i][j] += 1));
    |         ^^^^^^ help: you could try the macro: `loop_n`
error[E0433]: failed to resolve: use of undeclared type or module `KernelAvx`
   --> src/sgemm_kernel.rs:541:26
    |
541 |         let mut m = [[0; KernelAvx::NR]; KernelAvx::MR];
    |                          ^^^^^^^^^ use of undeclared type or module `KernelAvx`
error[E0433]: failed to resolve: use of undeclared type or module `KernelAvx`
   --> src/sgemm_kernel.rs:541:42
    |
541 |         let mut m = [[0; KernelAvx::NR]; KernelAvx::MR];
    |                                          ^^^^^^^^^ use of undeclared type or module `KernelAvx`
error[E0433]: failed to resolve: use of undeclared type or module `KernelAvx`
   --> src/dgemm_kernel.rs:820:27
    |
820 |             Alloc::new(n, KernelAvx::align_to()).init_with(elt)
    |                           ^^^^^^^^^ use of undeclared type or module `KernelAvx`
error: cannot find macro `loop_m!` in this scope
   --> src/dgemm_kernel.rs:855:9
    |
855 |         loop_m!(i, loop4!(j, m[i][j] += 1));
    |         ^^^^^^ help: you could try the macro: `loop_n`
error[E0433]: failed to resolve: use of undeclared type or module `KernelAvx`
   --> src/dgemm_kernel.rs:854:30
    |
854 |         let mut m = [[0; 4]; KernelAvx::MR];
    |                              ^^^^^^^^^ use of undeclared type or module `KernelAvx`
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -51,13 +51,19 @@ install:
 # the main build
 script:
   - |
+      set -e &&
+      set -v &&
       rustc --print cfg -Ctarget-cpu=native &&
-      cargo build --target=$TARGET &&
-      ([ -n "$BUILD_ONLY" ] || (
-      cargo test --target=$TARGET &&
-      cargo test --release --target=$TARGET &&
-      cargo doc --target=$TARGET &&
-      cargo bench --target=$TARGET ))
+      cargo build -v --target=$TARGET && (
+      if [ -n "$BUILD_ONLY" ]; then
+        cargo check --tests -v --target=$TARGET &&
+        cargo doc -v --target=$TARGET
+      else
+        cargo test -v --target=$TARGET &&
+        cargo test -v --release --target=$TARGET &&
+        cargo doc -v --target=$TARGET &&
+        cargo bench -v --target=$TARGET
+      fi)
 
 branches:
   only:

--- a/src/dgemm_kernel.rs
+++ b/src/dgemm_kernel.rs
@@ -814,10 +814,12 @@ mod tests {
     use super::*;
     use aligned_alloc::Alloc;
 
-    fn aligned_alloc<T>(elt: T, n: usize) -> Alloc<T> where T: Copy
+    fn aligned_alloc<K>(elt: K::Elem, n: usize) -> Alloc<K::Elem>
+        where K: GemmKernel,
+              K::Elem: Copy,
     {
         unsafe {
-            Alloc::new(n, KernelAvx::align_to()).init_with(elt)
+            Alloc::new(n, K::align_to()).init_with(elt)
         }
     }
 
@@ -827,8 +829,8 @@ mod tests {
         const K: usize = 4;
         let mr = K::MR;
         let nr = K::NR;
-        let mut a = aligned_alloc(1., mr * K);
-        let mut b = aligned_alloc(0., nr * K);
+        let mut a = aligned_alloc::<K>(1., mr * K);
+        let mut b = aligned_alloc::<K>(0., nr * K);
         for (i, x) in a.iter_mut().enumerate() {
             *x = i as _;
         }
@@ -850,6 +852,7 @@ mod tests {
     }
 
     #[test]
+    #[cfg(any(target_arch="x86", target_arch="x86_64"))]
     fn test_loop_m_n() {
         let mut m = [[0; 4]; KernelAvx::MR];
         loop_m!(i, loop4!(j, m[i][j] += 1));

--- a/src/sgemm_kernel.rs
+++ b/src/sgemm_kernel.rs
@@ -501,10 +501,12 @@ mod tests {
     use super::*;
     use aligned_alloc::Alloc;
 
-    fn aligned_alloc<T>(elt: T, n: usize) -> Alloc<T> where T: Copy
+    fn aligned_alloc<K>(elt: K::Elem, n: usize) -> Alloc<K::Elem>
+        where K: GemmKernel,
+              K::Elem: Copy,
     {
         unsafe {
-            Alloc::new(n, KernelAvx::align_to()).init_with(elt)
+            Alloc::new(n, K::align_to()).init_with(elt)
         }
     }
 
@@ -514,8 +516,8 @@ mod tests {
         const K: usize = 4;
         let mr = K::MR;
         let nr = K::NR;
-        let mut a = aligned_alloc(1., mr * K);
-        let mut b = aligned_alloc(0., nr * K);
+        let mut a = aligned_alloc::<K>(1., mr * K);
+        let mut b = aligned_alloc::<K>(0., nr * K);
         for (i, x) in a.iter_mut().enumerate() {
             *x = i as _;
         }
@@ -536,6 +538,7 @@ mod tests {
         test_a_kernel::<KernelFallback>("kernel");
     }
 
+    #[cfg(any(target_arch="x86", target_arch="x86_64"))]
     #[test]
     fn test_loop_m_n() {
         let mut m = [[0; KernelAvx::NR]; KernelAvx::MR];

